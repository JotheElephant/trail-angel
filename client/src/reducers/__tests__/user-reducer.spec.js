import userReducer from '../user-reducer';
import profileData from '../../../__tests__/fixtures/profile-data';
import actionTypes from '../../actions/action-types';

const initialState = {
  userId: null,
  email: null,
  nickname: null,
  avatarUrl: null,
  hasToken: false
};

describe('appReducer', () => {
  it('receives user data', () => {
    expect(userReducer(initialState,
      {
        type: actionTypes.RECEIVE_USER_DATA,
        profile: profileData
      })).toMatchSnapshot();
  });

  it('logs in a user', () => {
    expect(userReducer(initialState,
      {
        type: actionTypes.LOGIN_USER
      })).toMatchSnapshot();
  });

  it('logs out a user', () => {
    expect(userReducer(
      {
        initialState,
        hasToken: true
      },
      {
        type: actionTypes.LOGOUT_USER
      })).toMatchSnapshot();
  })
});