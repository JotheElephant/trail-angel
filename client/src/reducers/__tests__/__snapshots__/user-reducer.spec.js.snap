exports[`appReducer logs in a user 1`] = `
Object {
  "avatarUrl": null,
  "email": null,
  "hasToken": false,
  "nickname": null,
  "userId": null,
}
`;

exports[`appReducer logs out a user 1`] = `
Object {
  "hasToken": false,
  "initialState": Object {
    "avatarUrl": null,
    "email": null,
    "hasToken": false,
    "nickname": null,
    "userId": null,
  },
}
`;

exports[`appReducer receives user data 1`] = `
Object {
  "avatarUrl": "https://lh6.googleusercontent.com/-PHUnYsSh0kM/AAAAAAAAAAI/AAAAAAAAAQQ/Y5SZ33EhyYo/photo.jpg",
  "email": "wyatt.b.lindsey@gmail.com",
  "hasToken": true,
  "nickname": "wyatt.b.lindsey",
  "userId": "113879522060780032239",
}
`;
